### PR TITLE
Feature/be warmup

### PR DESCRIPTION
**Pull request by KostiukovAleksey**
- Basic entity Book was added.
- Controller for providing CRUD operations for Book.
- Some tests.

### CODE REVIEW

File: backend/nokia-innovative-project/src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# The SQL dialect makes Hibernate generate better SQL for the chosen database
+spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.Oracle12cDialect
+
+# Hibernate ddl auto (create, create-drop, validate, update)
+spring.jpa.hibernate.ddl-auto = update
+
+#spring.jpa.open-in-view is enabled by default. 
+#Therefore, database queries may be performed during view rendering. 
+#Explicitly configure spring.jpa.open-in-view to disable this warning
+spring.jpa.open-in-view=false
```
Comment (Dasiuss) [inline backend/nokia-innovative-project/src/main/resources/application.properties]: quick tip: you can use yaml for properties files! like:
```
spring.jpa:
    properties...
    hibernate....
```
less ... letters ?

File: backend/nokia-innovative-project/src/test/java/com/nokia/library/nokiainnovativeproject/repositories/RepositoriesTests.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import com.nokia.library.nokiainnovativeproject.repositories.BookRepository;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class RepositoriesTests {
+	@Autowired
+    private BookRepository bookRepository;
+	
+	@Test
+	public void whenAddBookThenBookAdded() {
```
Comment (MateuszPe) [inline backend/nokia-innovative-project/src/test/java/com/nokia/library/nokiainnovativeproject/repositories/RepositoriesTests.java]: There is no sense to load whole application context to test only repositories. 
Context should be used for e2e test or integration test. 

There is also no big sense to test repositories.  It is part of the code that was provided by Spring. I think that we can trust Spring developers. ;) 

You can write a test that will check your controller and database in one test. 
For example: 
test checks wheather after sending POST to endpoint `/api/v1/books` with body:
```
{
"title" : "book"
}
```
an appropriate entry in the database will be created after making the test

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/controllers/BookController.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import com.nokia.library.nokiainnovativeproject.entities.Book;
+import com.nokia.library.nokiainnovativeproject.exceptions.ResourceNotFoundException;
+import com.nokia.library.nokiainnovativeproject.repositories.BookRepository;
+
+@RestController
+@RequestMapping("/library")
+public class BookController {
+
+	@Autowired
+	private BookRepository bookRepository;
```
Comment (MateuszPe) [inline backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/controllers/BookController.java]: Please read about MVC and three-tier architecture. Then please refactor code to achieve this architecture.

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/config/profiles/AutotestsConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.nokia.library.nokiainnovativeproject.config.profiles;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+import javax.annotation.PostConstruct;
+
+@Configuration
+@Profile("autotests")
```
Comment (MateuszPe) [inline backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/config/profiles/AutotestsConfiguration.java]: This class is unnecessary.

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/config/profiles/DevelopmentConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.nokia.library.nokiainnovativeproject.config.profiles;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+import javax.annotation.PostConstruct;
+
+@Configuration
+@Profile("development")
```
Comment (MateuszPe) [inline backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/config/profiles/DevelopmentConfiguration.java]: This class is unnecessary.

File: backend/nokia-innovative-project/src/test/java/com/nokia/library/nokiainnovativeproject/controllers/BookControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@RequiredArgsConstructor
```
Comment (MateuszPe) [inline backend/nokia-innovative-project/src/test/java/com/nokia/library/nokiainnovativeproject/controllers/BookControllerTest.java]: I am sure that all tests fail.
Still you use default configuration and in default configuration you do not have any database configuration.

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Book.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.*;
+import java.io.Serializable;
+
+@Data
+@Builder
+@AllArgsConstructor
```
[the remark this answers is not in the document]
Comment (tubidubidam) [reply backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Book.java]: # sorry, missed that one :(
but now it's fixed

**Pull request merged**